ethereum/core/src/test/java/org/hyperledger/besu/ethereum/vm/operations/BeginSubOperationTest.java

```diff
@@ -85,17 +86,16 @@ public void shouldCalculateGasPrice() {
   }
 
   @Test
-  public void shouldNotConsumeStackItem() {
+  public void shouldHaltWithOutOfGasWhenBeginSubIsExecuted() {
     final BeginSubOperation operation = new BeginSubOperation(gasCalculator);
     final MessageFrame frame =
         createMessageFrameBuilder(Gas.of(1))
-            .pushStackItem(Bytes32.fromHexString("0xFF"))
-            .code(new Code(Bytes.fromHexString("0x6004b300b2b7")))
+            .pushStackItem(Bytes32.fromHexString("0x04"))
+            .code(new Code(Bytes.fromHexString("0x610400b2")))
             .returnStack(new ReturnStack(MessageFrame.DEFAULT_MAX_RETURN_STACK_SIZE))
```

Comment:
I guess this is a comment for another PR but why do we have the `ReturnStack` parameterized by size? The size of it is fixed in the EIP, yes? So we shouldn't expose that detail to the caller.

Reply:
yes you're right I always make this call `new ReturnStack (MessageFrame.DEFAULT_MAX_RETURN_STACK_SIZE)`  I will set this value on the ReturnStack object. I will do another PR to fix that

Reply:
Cool, then you can model it as a singleton class with a static `ReturnStack`, and private constructor.